index.bs

```diff
@@ -76,773 +93,795 @@ All <{iframe}> use cases are still valid and should continue to use <{iframe}>,
-  boolean sessionCookies;
-  boolean webSQL;
-};
-</xmp>
+Planned mitigation:
 
-<!-- ====================================================================== -->
-## Capture methods ## {#api-capture}
-<!-- ====================================================================== -->
+-  Controlled Frame will only be available within [[Isolated-Web-Apps|IWAs]]
 
-<!-- ====================================================================== -->
-## Event listener API ## {#api-event}
-<!-- ====================================================================== -->
+<b>An [[Isolated-Web-Apps|IWA]] may embed another [[Isolated-Web-Apps|IWA]] (or
+itself) via Controlled Frame to manipulate our [[Isolated-Web-Apps|IWA]]
+policies somehow (e.g. an Controlled Frame embedded [[Isolated-Web-Apps|IWA]]
+may detect it's being embedded due to the absence of the "controlledframe"
+policy-controlled feature).</b>
 
-<!-- ====================================================================== -->
-## Web Request API ## {#api-web-request}
-<!-- ====================================================================== -->
+Planned mitigation:
 
-<xmp class="idl">
-enum ResourceType {
-  "main_frame",
-  "sub_frame",
-  "stylesheet",
-  "script",
-  "image",
-  "font",
-  "object",
-  "xmlhttprequest",
-  "ping",
-  "csp_report",
-  "media",
-  "websocket",
-  "webbundle",
-  "other",
-};
+-  Controlled Frame can only point to "https" schemes, excluding the
+    "isolated-app" scheme used for [[Isolated-Web-Apps|IWAs]]
 
-callback interface WebRequestEventListener {
-  BlockingResponse? handleEvent(WebRequestEventDetails details);
-};
+<b>Controlled Frame could gain access to the powerful &lt;controlledframe&gt;
+element.</b>
 
-dictionary RequestFilter {
-  sequence<ResourceType> types;
-  sequence<USVString> urls;
-  long windowId;
-};
+An [[Isolated-Web-Apps|IWA]] that's not expected to use Controlled Frame may
+attempt to embed content.
 
-enum ExtraInfoSpec {
-  "asyncBlocking",
-  "blocking",
-  "extraHeaders",
-  "requestHeaders",
-  "responseHeaders",
-};
+Planned mitigation:
 
-[Exposed=Window, SecureContext]
-interface WebRequestEvent {
-  undefined addListener(WebRequestEventListener listener,
-                        optional RequestFilter filter = {},
-                        optional sequence<ExtraInfoSpec> extraInfoSpec);
-  boolean hasListener(WebRequestEventListener listener);
-  boolean hasListeners();
-  undefined removeListener(WebRequestEventListener listener);
-};
+-  Only embedder applications and their same-origin [[Isolated-Web-Apps|IWA]]
+    child navigables that have been granted the "controlledframe"
+    policy-controlled feature will be allowed access to the Controlled Frame
+    element.
+-  Same-origin child navigables without the "controlledframe"
+    policy-controlled feature will not be allowed to use a Controlled Frame
+    element (to be confirmed, see note below)
+-  Cross-origin iframes to [[Isolated-Web-Apps|IWAs]] can use the "allow"
```

Comment:
IWA APIs can never be delegated cross-origin.

Reply:
OK, I revised this section. PTAL.